keras_cv/metrics/object_detection/box_coco_metrics.py

```diff
@@ -192,17 +192,17 @@ def update_state_fn(self, y_true, y_pred, sample_weight=None):
         # Wrap the result function in a py_function and scope it to /cpu:0
         obj_result = obj.result
 
-        def result_on_host_cpu():
+        def result_on_host_cpu(force):
             with tf.device("/cpu:0"):
                 # Without the call to `constant` `tf.py_function` selects the
                 # first index automatically and just returns obj_result()[0]
-                return tf.constant(obj_result(), obj.dtype)
+                return tf.constant(obj_result(force), obj.dtype)
 
         obj.result_on_host_cpu = result_on_host_cpu
 
-        def result_fn(self):
+        def result_fn(self, force=False):
             py_func_result = tf.py_function(
-                self.result_on_host_cpu, inp=[], Tout=obj.dtype
+                self.result_on_host_cpu, inp=[force], Tout=obj.dtype
```

Comment:
Have you verified that this works in Graph mode on TPU?
Since force is a Python boolean, I would expect this to break, but perhaps I am misunderstanding

Reply:
This is invoked from python, so whatever value is passed will be converted to a Tensor.

Reply:
Also, this already doesn't work on TPU, right?